recipes/dd-handler.rb

```diff
@@ -26,7 +26,13 @@
 end
 
 include_recipe 'chef_handler'
-ENV['DATADOG_HOST'] = node['datadog']['url']
+
+# use either the site option or the url one, the latter having priority.
+node.run_state['dd_url'] = 'https://app.datadoghq.com'
+node.run_state['dd_url'] = 'https://app.' + node['datadog']['site'] unless node['datadog']['site'].nil?
+node.run_state['dd_url'] = node['datadog']['url'] unless node['datadog']['url'].nil?
+
+ENV['DATADOG_HOST'] = node.run_state['dd_url']
```

Comment:
I think setting `DATADOG_HOST` is needed only until `v0.9.0` of the handler ([`v0.10.0`](https://github.com/DataDog/chef-handler-datadog/releases/tag/v0.10.0)) added support for multiple endpoints, which required supporting the `url` config option directly on the handler)

It could be worth removing support for <v0.10.0 in the next major version of the cookbook. Until then, I'd prefer duplicated logic (and a `TODO` or `FIXME` comment) than storing stuff in the `run_state`